### PR TITLE
refactor: extra category rendering to separate class

### PR DESCRIPTION
**Pull request by patrickhulce**

### CODE REVIEW

File: lighthouse-core/test/report/v2/renderer/category-renderer-test.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    const detailsRenderer = new DetailsRenderer(dom);
+    renderer = new CategoryRenderer(dom, detailsRenderer);
+  });
+
+  after(() => {
+    global.URL = undefined;
+  });
+
+
+
```
Comment (brendankenny) [inline lighthouse-core/test/report/v2/renderer/category-renderer-test.js]: extra lines on purpose?

File: lighthouse-core/report/v2/report-template.html [hunk 2]
```diff
@@ -32,7 +32,8 @@
   <script>
     const dom = new DOM(document);
     const detailsRenderer = new DetailsRenderer(dom);
-    const renderer = new ReportRenderer(dom, detailsRenderer);
+    const categoryRenderer = new CategoryRenderer(dom, detailsRenderer);
```
Comment (brendankenny) [inline lighthouse-core/report/v2/report-template.html]: ha, we're going to need a map of these injections pretty soon

File: lighthouse-core/test/report/v2/renderer/category-renderer-test.js (hunk 1 above)
Comment (patrickhulce) [inline lighthouse-core/test/report/v2/renderer/category-renderer-test.js]: oops, nope fixed

File: lighthouse-core/report/v2/report-template.html (hunk 2 above)
Comment (patrickhulce) [inline lighthouse-core/report/v2/report-template.html]: ha, ya know I hear there's some newfangled way these days to state that a file `require`'s another...I wonder if that's something we could look into?

**Comment by brendankenny** [comment]
screen shot 2017-04-21 at 16 12 51

lol, thanks

**Comment by patrickhulce** [comment]
> Also, if you're willing to rebase #2000 for eric, then i think we can land this today. wdyt?

👍 I can live with that, share the rebase love ❤️

**Pull request merged**